Add en- and disable MMU to M709

3d-gussner opened the pull request:
Add X42 to erase the MMU eeprom

@gudnimg Any optimization possible?

Related to https://github.com/prusa3d/Prusa-Firmware/issues/4473 and https://github.com/prusa3d/Prusa-Link/issues/872

- [X] Update Doxygen
- [x] Update RepRap wiki
- [x] Update `M709` for MK4
   - [X]  `S0|0` and `X0|X1` already implemented
   - [x] Add `X42` to erase the MMU eeprom

3d-gussner committed:
Add en- and disable MMU to M709
Add X42 to erase the MMU eeprom

gudnimg commented on Firmware/Marlin_main.cpp:
```diff
@@ -8581,24 +8581,44 @@ SERIAL_PROTOCOLPGM("\n\n");
     {
+        if (code_seen('S'))
+        {
+            switch (code_value_uint8())
+            {
+            case 0:
+                MMU2::mmu2.Stop();
+                break;
+            case 1:
+                MMU2::mmu2.Start();
```
The EEPROM must be set accordingly here, otherwise the menus may show the wrong value e.g. `MMU [ON]` when it is really off.

3d-gussner replied:
@gudnimg Added eeprom update

gudnimg commented on Firmware/Marlin_main.cpp:
```diff
@@ -8581,24 +8581,44 @@ SERIAL_PROTOCOLPGM("\n\n");
 
     #### Parameters
-    - `X` - Reset MMU (0:soft reset | 1:hardware reset)
+    - `X` - Reset MMU (0:soft reset | 1:hardware reset | 42: erease MMU eeprom)
```
erease => erase

DRracer replied:
Also, it should be noted, that the new state of the MMU is stored in printer's EEPROM

3d-gussner replied:
@gudnimg fixed typo and @DRracer updated documentation

DRracer replied:
@3d-gussner : with the EEPROM comment I meant something like this:
"The new state of the MMU is stored in printer's EEPROM - i.e. if you disable the MMU via M709, it will not be activated after the printer resets."

3d-gussner replied:
@DRracer Better now? See https://github.com/prusa3d/Prusa-Firmware/pull/4474/commits/5cdb1916102fbac9bd5bace0509367911a6175e7

3d-gussner committed:
Update requested changes

3d-gussner committed:
Update doxygen for M709

gudnimg commented on Firmware/Marlin_main.cpp:
```diff
@@ -8609,10 +8633,14 @@ SERIAL_PROTOCOLPGM("\n\n");
                 break;
             }
         }
+        printf_P(_n("MMU state:%d\n"), MMU2::mmu2.Enabled());
```
@3d-gussner the PR looks good to me now. But I’m not sure about this log.

I feel it is not descriptive enough.

Maybe `MMU On` and `MMU Off` is better? Then it would be similar to the log I added for SpoolJoin when the printer boots up.

3d-gussner committed:
Updated MMU serial status message
- Shown during bootup
- Shown on LCD menue MMU [ON|OFF]
- Shown using `M709`
- Shown when `Disable`d with MMU ERROR FW UPDATE NEEDED

3d-gussner committed:
Update doxygen M709

gudnimg reviewed (approved):
Tested on MK3S+ with MMU3.
